File uploads failing in ChatViewController (ticket #7158)

## What did you do?

Tried to upload an image from the phone, using the default Stream Chat CDN

## What did you expect to happen?

Image to succesfully upload, and render in the chat

## What happened instead?

The screen cuts off the previous chat message. On closing and going back to the chat, the chat appears normal again but the image isn't visible.

## GetStream Environment

GetStream Chat version: 2.4.1 
**iOS version:**14.1 (reproducible on multiple OS versions)
**Swift version:**5.3 
**Xcode version:**12.1
**Device:**iPhone X (reproducible on multiple devices)

## Additional context

We also opened this with support, StreamChat ticket #7158

Stack trace from an upload:

*[11 Nov 13:41:38.478] [DEBUG] Preparing a POST request to /channels/livestream/[removed]/image...
[11 Nov 13:41:38.478] [DEBUG] ⏫ Uploading 2327207 bytes
[11 Nov 13:41:38.478] [DEBUG] ➡️ POST https://chat-proxy-us-east.stream-io-api.com/channels/livestream/[removed]/image?api_key=[removed]&connection_id=[removed]
[11 Nov 13:41:38.478] [DEBUG] URL query items:
▫️ connection_id=[removed]
[11 Nov 13:41:38.478] [DEBUG] ￼ Uploading 2272 KB data...
[11 Nov 13:41:38.582] [DEBUG] [7] 1048576/2327207, 45.1%
[11 Nov 13:41:38.787] [DEBUG] [7] 2327207/2327207, 100.0%
[11 Nov 13:41:39.080] [DEBUG] ⏱ URL request to /channels/livestream/[removed]/image finished in 0.602 seconds.
[11 Nov 13:41:39.080] [DEBUG] ⬅️ Response 201 (181 bytes): https://chat-proxy-us-east.stream-io-api.com/channels/livestream/[removed]/image?api_key=[removed]&connection_id=[removed]
[11 Nov 13:41:39.080] [INFO] ￼ {
  "duration" : "186.92ms",
  "file" : "https:\/\/stream-chat-us-east-c7.imgix.net\/97821\/images\/[removed].IMG_6190.JPG?ro=0&s=[removed]"
}
[11 Nov 13:41:39.081] [DEBUG] ⏱ Parsing the JSON response from /channels/livestream/[removed]/image finished in 0.000 seconds.
[11 Nov 13:41:39.787] [DEBUG] Preparing a POST request to /channels/livestream/[removed]/message...
[11 Nov 13:41:39.788] [DEBUG] ➡️ POST https://chat-proxy-us-east.stream-io-api.com/channels/livestream/[removed]/message?api_key=[removed]&connection_id=[removed]
[11 Nov 13:41:39.788] [DEBUG] URL query items:
▫️ connection_id=[removed]
[11 Nov 13:41:39.788] [ERROR] ￼ Request Body {
  "message" : {
    "text" : "",
    "silent" : false
  }
}
[11 Nov 13:41:40.039] [DEBUG] ⏱ URL request to /channels/livestream/[removed]/message finished in 0.250 seconds.
[11 Nov 13:41:40.039] [DEBUG] ⬅️ Response 201 (693 bytes): https://chat-proxy-us-east.stream-io-api.com/channels/livestream/[removed]/message?api_key=[removed]&connection_id=[removed]*

gz#7309

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### 🐞 Fixed
 - `MembersQueryResponse.member` is now public [#574](https://github.com/GetStream/stream-chat-swift/issues/574)
 - Fix plus signs in User fields becoming spaces in the set user request, preventing names with + signs, base64 custom fields, etc. [#572](https://github.com/GetStream/stream-chat-swift/issues/572)
+- Fix images and files are not being send. This was a bug introduced in 2.4.1, now resolved [#585](https://github.com/GetStream/stream-chat-swift/issues/585)
 
 # [2.4.1](https://github.com/GetStream/stream-chat-swift/releases/tag/2.4.1)
 _October 23, 2020_

--- a/Sources/UI/Chat/Chat View Controller/ChatViewController+Composer.swift
+++ b/Sources/UI/Chat/Chat View Controller/ChatViewController+Composer.swift
@@ -189,10 +189,6 @@ extension ChatViewController {
             composerEditingContainerView.animate(show: false)
         }
         
-        // We don't want users to send the same message multiple times
-        // in case their internet is slow and message isn't sent immediately
-        composerView.reset()
-        
         presenter?.rx.send(text: text,
                            showReplyInChannel: composerView.alsoSendToChannelButton.isSelected,
                            parseMentionedUsers: parseMentionedUsersOnSend)
@@ -206,6 +202,10 @@ extension ChatViewController {
                     self?.show(error: $0)
                 })
             .disposed(by: disposeBag)
+        
+        // We don't want users to send the same message multiple times
+        // in case their internet is slow and message isn't sent immediately
+        composerView.reset()
     }
     
     private func findCommand(in text: String) -> String? {

